runtime/vam/agg.go

```diff
@@ -110,6 +153,7 @@ func NewSum(zctx *zed.Context, parent Puller, name string) *Sum {
 		parent: parent,
 		zctx:   zctx,
 		name:   name,
+		field:  NewDotExpr(zctx, &This{}, name),
```

Comment:
Nit: Initialize in declaration order for reader's benefit.
```suggestion
		field:  NewDotExpr(zctx, &This{}, name),
		name:   name,
```